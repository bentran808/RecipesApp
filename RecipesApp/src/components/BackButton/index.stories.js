import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react-native';
import React from 'react';
import { BackArrow, MenuIcon } from 'theme';
import BackButton from '.';
import CenterView from 'storybook/stories/CenterView';

storiesOf('Back Button', module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add('with back icon', () => <BackButton source={BackArrow} onPress={action('clicked-icon')} />)
  .add('with menu icon', () => <BackButton source={MenuIcon} onPress={action('clicked-icon')} />);
