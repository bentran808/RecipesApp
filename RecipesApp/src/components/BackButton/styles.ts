import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  btnContainer: {
    alignItems: 'center',
    borderRadius: 180,
    padding: 8,
    margin: 10,
    backgroundColor: 'white',
    shadowColor: 'black',
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 3
  },
  btnIcon: {
    height: 17,
    width: 17
  },
  badge: {
    position: 'absolute',
    right: 0,
    top: 0,
    backgroundColor: 'red',
    borderRadius: 8,
    width: 16,
    height: 16,
    justifyContent: 'center',
    alignItems: 'center'
  }
});

export default styles;
