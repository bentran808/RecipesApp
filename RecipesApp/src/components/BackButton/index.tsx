import React from 'react';
import { Image, TouchableHighlight } from 'react-native';
import { BackArrow } from 'theme';

import styles from './styles';

type Props = {
  onPress: () => void;
};

const BackButton = ({ onPress }: Props) => {
  return (
    <TouchableHighlight onPress={onPress} style={styles.btnContainer}>
      <Image source={BackArrow} style={styles.btnIcon} />
    </TouchableHighlight>
  );
};

export default React.memo(BackButton);
