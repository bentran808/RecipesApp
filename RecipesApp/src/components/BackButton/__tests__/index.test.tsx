import BackButton from 'components/BackButton';
import renderer from 'react-test-renderer';

describe('Back Button Component', () => {
  test('should render correctly', () => {
    const tree = renderer.create(<BackButton onPress={jest.fn()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
