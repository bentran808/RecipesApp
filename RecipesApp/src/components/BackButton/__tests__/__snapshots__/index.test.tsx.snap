// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back Button Component should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 180,
      "elevation": 3,
      "flex": 1,
      "margin": 10,
      "padding": 8,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 17,
        "width": 17,
      }
    }
  />
</View>
`;
